inc/Engine/Optimization/RUCSS/Warmup/ResourceFetcherProcess.php

```diff
@@ -0,0 +1,114 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Engine\Optimization\RUCSS\Warmup;
+
+use WP_Rocket\Engine\Optimization\RUCSS\Database\Queries\ResourcesQuery;
+use \WP_Rocket_WP_Background_Process;
+
+class ResourceFetcherProcess extends WP_Rocket_WP_Background_Process {
+
+	/**
+	 * Background process action name.
+	 *
+	 * @var string
+	 */
+	protected $action = 'rucss_warmup_resource_fetcher';
+
+	/**
+	 * Background process prefix.
+	 *
+	 * @var string
+	 */
+
+	protected $prefix = 'rocket';
+
+	/**
+	 * ResourcesQuery instance.
+	 *
+	 * @var ResourcesQuery
+	 */
+	private $resources_query;
+
+	/**
+	 * ResourceFetcherProcess constructor.
+	 *
+	 * @param ResourcesQuery $resources_query ResourcesQuery instance.
+	 */
+	public function __construct( ResourcesQuery $resources_query ) {
+		parent::__construct();
+
+		$this->resources_query = $resources_query;
+	}
+
+
+	/**
+	 * Do the task for each page resources.
+	 *
+	 * @param array $resources Array of collected resources on the page.
+	 *
+	 * @return false
+	 */
+	protected function task( $resources ) {
+		if ( ! is_array( $resources ) ) {
+			return false;
+		}
```

Comment:
In php 7 we can specify array type in signature and get rid of the type checking.
Better yet, since the parent class seems to require an iterable for `task()` (@see the description in the docblock), it might be better to change the signature to `abstract protected function task( iterable $item ) there rather than a mixed/unspecified.